docs/content/concepts/ops-jobs-graphs/ops.mdx

```diff
@@ -1,42 +1,41 @@
+| <PyObject object="OpExecutionContext"/> | An object exposing Dagster system APIs for resource access, logging, and more. Can be injected into an op by specifying `context` as the first argument of the compute function.                           |
+| <PyObject object="OpDefinition"  />     | Class for ops. You will rarely want to instantiate this class directly. Instead, you should use the <PyObject object="op" decorator />. |
 
 ## Overview
 
-Ops are used to define computations. Ops can later be assembled into [Jobs](/concepts/ops-jobs-graphs/jobs-graphs). Ops generally perform one specific action and are used for batch computations. For example, you can use an op to:
+Ops are Dagster's atomic unit of computation. Individual ops should perform relatively simple tasks. Collections of ops can then be assembled into [Jobs or Graphs](/concepts/ops-jobs-graphs/jobs-graphs) to perform more complex tasks. Some examples of tasks appropriate for a single op:
 
-- Derive a data set from some other data sets.
+- Add computed columns to a dataframe.
 - Execute a database query.
 - Initiate a Spark job in a remote cluster.
 - Query an API and store the result in a data warehouse.
 - Send an email or Slack message.
 
-Depending on how you execute a job, Dagster can run all the ops in the job within the same process or isolate them in different processes.
+The op as computational unit enables many useful features for data orchestration:
 
-Ops have several important properties:
-
-- **Inputs and Outputs**: Ops have defined [inputs and outputs](#inputs-and-outputs), which can be optionally typed. These types are validated at runtime.
-- **Configurable**: Ops can be configured, using a strongly typed [configuration system](/concepts/configuration/config-schema).
-- **Emit an Event Stream**: Ops can emit a stream of structured events, such as AssetMaterializations. These events can be viewed in [Dagit](/concepts/dagit/dagit), Dagster's UI tool.
-- **Individually testable**: See the [testing page](/concepts/testing) for more detail.
+- **Flexible execution strategies**: Data pipelines are frequently developed locally and later deployed to production. Ops represent sealed units of logic independent of pipeline execution strategy, making the transition from development to production painless. Collections of ops (i.e. [Jobs](/concepts/ops-jobs-graphs/jobs-graphs)) can be passed to an appropriate [Executor](/deployment/executors) for single-process execution or distribution across a cluster.
+- **Input and output management**: Ops have defined [inputs and outputs](#inputs-and-outputs) analogous to the arguments and return value(s) of a Python function. However, op inputs/outputs can be more richly annotated than Python function arguments. This annotation can instruct Dagster's engines how to validate, cache, serialize, and otherwise manage data as it flows between ops.
```

Comment:
I think replacing the last sentence with something more specific could make this bullet more clear.  E.g. something like "Annotating inputs/outputs with Dagster Types instructs Dagster on how to validate them, and annotating them with io_manager_keys instructs Dagster on how to serialize them." Would also omit "engines" to avoid risking users thinking that an "engine" is a meaningful abstraction in Dagster.

Reply:
Will adjust last sentence.

Thx for the pointer on "engine", I can delete that. But this brings up a point I'm hazy on-- is there a central entity that coordinates job execution, resource access, etc? What is it called? I have seen a few places where different terms are used:

- "An optional first argument to a op's compute function, providing access to Dagster system information such as resources, logging, and the underlying **Dagster instance**."
- "Yielding events from within the body of an op is a useful way of communicating with the **Dagster framework**."
- "It is also possible to raise Dagster-specific exceptions, which will indicate to the **framework** to halt the op execution entirely and perform some action."
-"The **Dagster Daemon** operates schedules, sensors, and run queuing. Currently, replicas are not supported." 
- "ENGINE EVENT" is a DagsterEventType that shows up in Dagit.

"Framework" here seems wrong, as it is not a specific entity that communicates with other entities. My guess is "Dagster Instance" is best (does this mean an instance of the Daemon?), but this term is barely used in the docs.

Reply:
Dagster Instance is a specific concept in Dagster: https://docs.dagster.io/deployment/dagster-instance
In this case, I usually would say "This annotation can instruct Dagster **machinery** how to validate, cache, serialize, and otherwise manage data as it flows between ops." - "machinery" is the word we use in code comments and docstring for this case.

Reply:
Good suggestion on "machinery". And looks like I missed that Instance page, I think because I haven't paid enough attention to "Deployment" section.